lib/puppet/provider/package/gem.rb

```diff
@@ -108,6 +108,15 @@ def install(useversion = true)
     self.fail "Could not install: #{output.chomp}" if output.include?("ERROR")
   end
 
+  # Allow Gem version expressions in ensure
+  def insync?(is)
+    begin
+      Gem::Dependency.new('', resource[:ensure]).match?('', is)
+    rescue
+      false
```

Comment:
Looking at this, will it work correctly with "present" or "absent" in the ensure? That would cause the `Gem::Dependency.new` to throw an exception, which would always be false... but I also know that ensure absest/present have some weird special code paths.

Basically I have no idea what would happen here in that case. I'm going to play with this code before next week's PR triage meeting so we can decide what to do with this.